src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestAttributeValueResolver.php

```diff
@@ -27,7 +27,13 @@
      */
     public function supports(Request $request, ArgumentMetadata $argument)
     {
-        return !$argument->isVariadic() && $request->attributes->has($argument->getName());
+        if ($argument->isVariadic()) {
+            return false;
+        }
+
+        $argumentName = $argument->getName();
+
+        return $request->attributes->has($argumentName) && null !== $request->attributes->get($argumentName);
```

Comment:
@xabbuh I agree on that point. If the compiled route would be without the default value, everything will work as expected